File: LeetCode Mastery/1. Two Sum/two-sum/src/Solution_One.java
import java.util.Arrays;

/*
 * Solution 1
 * - double pass
 * - O(n^2)
 * - runtime: 45 ms
 * - memory: 44.90 MB
 */
public class Solution_One {
    public Solution_One(int[] input_nums, int input_target) {
        System.out.println("\nSolution 1 -->");
        System.out.println("nums[]: " + Arrays.toString(input_nums));
        System.out.println("target: " + input_target);
        System.out.println("output: " + Arrays.toString(twoSum(input_nums, input_target)));
    }
    
    private int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[j] == complement && i != j) {
                    return new int[] {i, j};
                }
            }
        }
        return null;
    }
}

File: LeetCode Mastery/1. Two Sum/two-sum/bin/Solution_One.py
# Solution 1
# - double pass
# - O(n^2)
# - runtime: 2017 ms
# - memory: 17.56 MB

from typing import List

class Solution_One:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        for i in range(n):
            complement = target - nums[i]
            for j in range(n):
                if (nums[j] == complement and i != j):
                    return [i, j]

    nums = [2, 7, 11, 15]
    target = 9
    print(twoSum(self = True, nums = nums, target = target))